lib/tasks/seek_upgrades.rake

```diff
@@ -201,7 +202,43 @@ namespace :seek do
       PublicationType.find_or_initialize_by(key: "diplomthesis").update(title:"Diplom Thesis", key: "diplomthesis")
       puts 'Add new type '+PublicationType.find_by(key:"diplomthesis").title
     end
-
   end
 
+  task(convert_roles: [:environment]) do
+    puts 'Converting roles...'
+    disable_authorization_checks do
+      Person.find_each do |person|
+        RoleType.for_system.each do |rt|
+          mask = rt.id
+          if person.roles_mask & mask == mask
```

Comment:
I'm also a little unclear how the mask relates to the rt.id ?

Reply:
It's doing the bitwise `&` before the `==`, so it's `(person.roles_mask & mask) == mask`. Could just be `!= 0` instead..

I set the ID of each `RoleType` to be the same as the mask of the old role in case it is ever needed again